compiler/crates/relay-typegen/src/flow.rs

```diff
@@ -14,6 +14,20 @@ pub struct FlowPrinter {
 }
 
 impl Writer for FlowPrinter {
+    fn write_ast(&mut self, ast: &AST) -> String {
```

Comment:
We should keep the previous way of not generating intermediate `String`s. What this implementation is doing is:
1. Allocate a new `String`.
2. Write into that string.
3. Write the result onto the larger result.

The updated version just directly wrote to the target, skipping the intermediate string.

I think this probably got lost in a rebase you did, as I updated this recently.


The other lint fixes in this file (`&[StringKey]`, instead of `&Vec<StringKey>` are still good)

Reply:
Whoops yeah, that got lost in either a merge or rebase. Fixed.